--- a/ufo2ft/outlineCompiler.py
+++ b/ufo2ft/outlineCompiler.py
@@ -983,7 +983,14 @@
         topDict.UnderlineThickness = otRound(underlineThickness)
         # populate font matrix
         unitsPerEm = otRound(getAttrWithFallback(info, "unitsPerEm"))
-        topDict.FontMatrix = [1.0 / unitsPerEm, 0, 0, 1.0 / unitsPerEm, 0, 0]
+        topDict.FontMatrix = [
+            round(1.0 / unitsPerEm, 14),
+            0,
+            0,
+            round(1.0 / unitsPerEm, 14),
+            0,
+            0
+        ]
         # populate the width values
         if not any(hasattr(info, attr) and getattr(info, attr) is not None
                    for attr in ("postscriptDefaultWidthX",
